exasol_integration_test_docker_environment/lib/api/spawn_test_environment.py

```diff
@@ -0,0 +1,81 @@
+    SpawnTestEnvironmentWithDockerDB
+
+
+def spawn_test_environment(
+        environment_name: str,
+        database_port_forward: Optional[int] = None,
+        bucketfs_port_forward: Optional[int] = None,
+        db_mem_size: str = "2 GiB",
+        db_disk_size: str = "2 GiB",
+        nameserver: Tuple[str,...] = tuple(),
+        deactivate_database_setup: bool = False,
+        docker_runtime: Optional[str] = None,
+        docker_db_image_version: str = LATEST_DB_VERSION,
+        docker_db_image_name: str = "exasol/docker-db",
+        create_certificates: bool = False,
+        source_docker_repository_name: str = 'exasol/script-language-container',
+        source_docker_tag_prefix: str = '',
+        source_docker_username: Optional[str] = None,
+        source_docker_password: Optional[str] = None,
+        target_docker_repository_name: str = 'exasol/script-language-container',
+        target_docker_tag_prefix: str = '',
+        target_docker_username: Optional[str] = None,
+        target_docker_password: Optional[str] = None,
+        output_directory: str = ".build_output",
+        temporary_base_directory: str = "/tmp",
+        workers: int = 5,
+        task_dependencies_dot_file: Optional[str] = None) -> bool:
+    """
+    This command spawn a test environment with a docker-db container and a connected test-container.
+    The test-container is reachable by the database for output redirects of UDFs.
+    """
+    parsed_db_mem_size = humanfriendly.parse_size(db_mem_size)
+    if parsed_db_mem_size < humanfriendly.parse_size("1 GiB"):
+        raise ArgumentConstraintError("db_mem_size", "needs to be at least 1 GiB")
+    parsed_db_disk_size = humanfriendly.parse_size(db_disk_size)
+    if parsed_db_disk_size < humanfriendly.parse_size("100 MiB"):
+        raise ArgumentConstraintError("db_disk_size", "needs to be at least 100 MiB")
+    set_build_config(False,
+                     tuple(),
+                     False,
+                     False,
+                     output_directory,
+                     temporary_base_directory,
+                     None,
+                     None)
+    set_docker_repository_config(source_docker_password, source_docker_repository_name, source_docker_username,
+                                 source_docker_tag_prefix, "source")
+    set_docker_repository_config(target_docker_password, target_docker_repository_name, target_docker_username,
+                                 target_docker_tag_prefix, "target")
+    task_creator = lambda: generate_root_task(task_class=SpawnTestEnvironmentWithDockerDB,
+                                              environment_name=environment_name,
+                                              database_port_forward=str(
+                                                  database_port_forward) if database_port_forward is not None else None,
```

Comment:
Do we have the ports defined as string parameters in the tasks?

Reply:
Yes, [here](https://github.com/exasol/integration-test-docker-environment/blob/92772852d77f19e260c83ad27dafdf5ea9aaeafc/exasol_integration_test_docker_environment/lib/test_environment/spawn_test_database.py#L94) they are casted back to `int`.

Reply:
hm, a bit not so elegant, it might be because IntParameter can't be None,maybe something we should check

Reply:
Open #239